setupModuleAliases.js

```diff
@@ -154,25 +148,30 @@ const addModuleAliases = (
 	}
 )
 
-const getAliasList = (
-	basePath => (
-		require(
-			basePath
-			.concat('/package.json')
-		)
-		._moduleAliases
-	)
-)
-
-const setupModuleAliases = (
-	basePath => {
-		addModuleAliases(
-			basePath,
-			getAliasList(
-				basePath
-			),
-		)
-	}
-)
-
-module.exports = setupModuleAliases
+const getAliasesFromPackageJson = (basePath) => {
+  const packageJsonPath = basePath.concat("/package.json");
+  const packageJson = require(packageJsonPath);
+  if (!packageJson)
+    throw new Error(`No package.json could be located at the path: ${packageJsonPath}
+	This path is determined based on the require('better-module-alias')('PATH-GOES-HERE'). It is recommended you use require('better-module-alias')(__dirname)`);
+
+  const moduleAliases = packageJson._moduleAliases;
+  if (!moduleAliases)
+    throw new Error(
+      'No module aliases could be found in the package.json. Please add a "_moduleAliases" property to the package.json.'
+    );
+
+  return moduleAliases;
+};
+
+const setupModuleAliases = (basePath, functionImports) => {
+  if (functionImports) {
+    var aliases = functionImports;
+  } else {
+    var aliases = getAliasesFromPackageJson(basePath);
+  }
+
+  addModuleAliases(basePath, aliases);
+};
```

Comment:
Can we put this in a ternary in the second arg?

Or do `let aliases` on line `168` and `aliases =` on `169` and `171`. Rather than defining the variable in the `if` and relying on hoisting.

Reply:
The ternary looked kind of weird, so I just used a let.